### PR TITLE
termbox: 1.1.2 -> 1.1.3

### PR DESCRIPTION
**Pull request by adsr**
Repointing repo to termbox/termbox as nsf/termbox is no longer maintained.

###### Motivation for this change

Update version and repo location

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by hugolgst — APPROVED** [review]
Looks good to me.

Result of `nixpkgs-review pr 99581`:
```
3 packages built:
mle nimmm termbox
```

**Comment by fgaz** [comment]
How do we know that's the new canonical/established location?

**Comment by fgaz** [comment]
I looked around a bit and

* freebsd ports just moved it to the new repo: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=249672
* termbox-haskell was moved under github.com/termbox by its author
* the new termbox/termbox maintainer already contributed to he project before the fork and maintains a text editor based on termbox that's packaged on many distros

So I guess that's enough for trust, but it doesn't seem to be that established yet (for example of the two dependent packages in nixpkgs, only one points to the fork)

**Comment by adsr** [comment]
Hi @fgaz yes I am working to recentralize development at termbox/termbox and avoid fork hell. I updated FreeBSD last week and will be updating Alpine and Debian next. If you'd like I can send you a copy of the email I sent to the termbox community a few months ago.

**Comment by fgaz** [comment]
Hi @adsr, thanks for chiming in (and for your work on a new central source of course)!

> If you'd like I can send you a copy of the email I sent to the termbox community a few months ago

If it isn't a problem please do

File: pkgs/development/libraries/termbox/default.nix [hunk 1]
```diff
@@ -1,31 +1,22 @@
-{ stdenv, fetchFromGitHub, python3, wafHook, fetchpatch }:
+{ stdenv, fetchFromGitHub, fetchpatch }:
```
Comment (fgaz) [inline pkgs/development/libraries/termbox/default.nix:1]: `fetchpatch` shouldn't be needed anymore

**Review by fgaz — CHANGES_REQUESTED** [review]

**Comment by adsr** [comment]
@fgaz :+1: I forwarded a copy to the email address on your github profile.

Thanks for the review.

**Review by fgaz — APPROVED** [review]
LGTM

Also @adsr I didn't even realize you also authored this pr, so that's the reason for my previous comments' weirdness like using third person :-P

**Comment by adsr** [comment]
Hello. Can someone add a `hacktoberfest-accepted` label to this issue so that it counts toward Hacktoberfest?

**Comment by fgaz** [comment]
I have no power to do that, sorry

(But I do have some hacktoberfest-tagged repos 🙂)

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>mle</li>
    <li>nimmm</li>
    <li>termbox</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Fails to build on darwin:
```
these derivations will be built:
  /nix/store/qdh2rvwkm08sln5hq90wn21a3s8662ps-termbox-1.1.3.drv
building '/nix/store/qdh2rvwkm08sln5hq90wn21a3s8662ps-termbox-1.1.3.drv'...
unpacking sources
unpacking source archive /nix/store/nm569bqm38wnn9wdwl72ykwdk5jflrzc-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/k89nm2jva0qmvd970f84wq2iq1iwm9bs-bash-4.4-p23/bin/bash prefix=/nix/store/d4lqkg71vi739sg9b8a5ybfdrkljx33h-termbox-1.1.3
clang -c -std=c99 -Wall -Wextra -pedantic -Wno-unused-result -fPIC -g -O3 -D_XOPEN_SOURCE  termbox.c -o termbox.o
clang -c -std=c99 -Wall -Wextra -pedantic -Wno-unused-result -fPIC -g -O3 -D_XOPEN_SOURCE  utf8.c -o utf8.o
ar rcs libtermbox.a termbox.o utf8.o
clang -shared -Wl,-h,libtermbox.so.1 termbox.o utf8.o -o libtermbox.so.1.0.0
ld: unknown option: -h
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:20: libtermbox.so.1.0.0] Error 1
builder for '/nix/store/qdh2rvwkm08sln5hq90wn21a3s8662ps-termbox-1.1.3.drv' failed with exit code 2
error: build of '/nix/store/qdh2rvwkm08sln5hq90wn21a3s8662ps-termbox-1.1.3.drv' failed
```

**Comment by adsr** [comment]
Hello, can someone try building with this patch?

https://github.com/termbox/termbox/commit/4235e57bb447b7df053cc07354577f406476c646

I don't have a mac to test with. If it looks ok I can tag as 1.1.4 or add a patch here.

**Comment by SuperSandro2000** [comment]
> Hello, can someone try building with this patch?
> 
> [termbox/termbox@4235e57](https://github.com/termbox/termbox/commit/4235e57bb447b7df053cc07354577f406476c646)
> 
> I don't have a mac to test with. If it looks ok I can tag as 1.1.4 or add a patch here.

I don't have the time to test this. Can you just mark it broken on darwin?

**Comment by adsr** [comment]
Build passes on a fastmac vm[1]

https://github.com/fastai/fastmac

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 99581` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>mle: log was empty</li>
<li>nimmm: log was empty</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 99581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>mle</li>
<li>nimmm</li>
<li>termbox</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
darwin:
```
builder for '/nix/store/z2hb4y38h0iaym72wy21hma4pq3kpmci-termbox-1.1.3.drv' failed with exit code 2; last 10 log lines:                                                                       no configure script, doing nothing                                                                                                                                                          building                                                                                                                                                                                    build flags: SHELL=/nix/store/4n2ah3plbfnfz14wrzmz1pkfw78y9kz4-bash-4.4-p23/bin/bash prefix=/nix/store/frn8mfax71715zlipw8zxchy9ka20lh2-termbox-1.1.3
  clang -c -std=c99 -Wall -Wextra -pedantic -Wno-unused-result -fPIC -g -O3 -D_XOPEN_SOURCE  termbox.c -o termbox.o
  clang -c -std=c99 -Wall -Wextra -pedantic -Wno-unused-result -fPIC -g -O3 -D_XOPEN_SOURCE  utf8.c -o utf8.o
  ar rcs libtermbox.a termbox.o utf8.o
  clang -shared -Wl,-h,libtermbox.so.1 termbox.o utf8.o -o libtermbox.so.1.0.0
  ld: unknown option: -h
  clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
  make: *** [Makefile:20: libtermbox.so.1.0.0] Error 1
```

**Comment by fgaz** [comment]
I see that 1.1.4 is out

**Comment by adsr** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/113831
